drivers/retained_mem/retained_mem_silabs_buram.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2023, Nordic Semiconductor ASA
+ * Copyright (c) 2023, Bjarki Arge Andreasen
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_buram
+
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_silabs.h>
+#include <zephyr/drivers/retained_mem.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(retained_mem_silabs_buram, CONFIG_RETAINED_MEM_LOG_LEVEL);
+
+struct silabs_buram_data {
+#ifdef CONFIG_RETAINED_MEM_MUTEXES
+	struct k_mutex lock;
+#endif
+};
+
+struct silabs_buram_config {
+	uint8_t *address;
+	size_t size;
+	const struct device *clock_dev;
```

Comment:
@nordicjm I wonder whether the default `retained_mem_zephyr_ram` could be updated to include clock/initialization feature. That is the only diff I can spot here (or I am missing anything?)

Reply:
When it comes to doing clock init in the generic driver: This will be difficult, since clock_control uses a SoC-specific data structure `clock_control_subsys_t`, and there is no generic macro to instantiate this structure (no equivalent to `pinctrl_soc.h` to draw a parallel to pinctrl).